File: pageindicatorview/src/main/java/com/rd/draw/data/PositionSavedState.java
package com.rd.draw.data;

import android.os.Parcel;
import android.os.Parcelable;
import android.view.View;

public class PositionSavedState extends View.BaseSavedState {

    private int selectedPosition;
    private int selectingPosition;
    private int lastSelectedPosition;

    public PositionSavedState(Parcelable superState) {
        super(superState);
    }

    private PositionSavedState(Parcel in) {
        super(in);
        this.selectedPosition = in.readInt();
        this.selectingPosition = in.readInt();
        this.lastSelectedPosition = in.readInt();
    }

    public int getSelectedPosition() {
        return selectedPosition;
    }

    public void setSelectedPosition(int selectedPosition) {
        this.selectedPosition = selectedPosition;
    }

    public int getSelectingPosition() {
        return selectingPosition;
    }

    public void setSelectingPosition(int selectingPosition) {
        this.selectingPosition = selectingPosition;
    }

    public int getLastSelectedPosition() {
        return lastSelectedPosition;
    }

    public void setLastSelectedPosition(int lastSelectedPosition) {
        this.lastSelectedPosition = lastSelectedPosition;
    }

    @Override
    public void writeToParcel(Parcel out, int flags) {
        super.writeToParcel(out, flags);
        out.writeInt(this.selectedPosition);
        out.writeInt(this.selectingPosition);
        out.writeInt(this.lastSelectedPosition);
    }

    public static final Parcelable.Creator<PositionSavedState> CREATOR = new Parcelable.Creator<PositionSavedState>() {
        public PositionSavedState createFromParcel(Parcel in) {
            return new PositionSavedState(in);
        }

        public PositionSavedState[] newArray(int size) {
            return new PositionSavedState[size];
        }
    };
}

File: pageindicatorview/src/main/java/com/rd/draw/data/PositionSavedState.kt
package com.rd.draw.data

import android.os.Parcel
import android.os.Parcelable
import android.view.View

class PositionSavedState : View.BaseSavedState {

	var selectedPosition: Int = 0
	var selectingPosition: Int = 0
	var lastSelectedPosition: Int = 0

	constructor(superState: Parcelable) : super(superState) {}

	private constructor(`in`: Parcel) : super(`in`) {
		this.selectedPosition = `in`.readInt()
		this.selectingPosition = `in`.readInt()
		this.lastSelectedPosition = `in`.readInt()
	}

	override fun writeToParcel(out: Parcel, flags: Int) {
		super.writeToParcel(out, flags)
		out.writeInt(this.selectedPosition)
		out.writeInt(this.selectingPosition)
		out.writeInt(this.lastSelectedPosition)
	}

	companion object {

		@JvmField
		val CREATOR: Parcelable.Creator<PositionSavedState> = object : Parcelable.Creator<PositionSavedState> {
			override fun createFromParcel(`in`: Parcel): PositionSavedState {
				return PositionSavedState(`in`)
			}

			override fun newArray(size: Int): Array<PositionSavedState?> {
				return arrayOfNulls(size)
			}
		}
	}
}
